### PR TITLE
Add cooked data read tracing

### PR DESCRIPTION
**Pull request by depletionmode**
## Summary of the Pull Request

Added trace to conhost to instrument buffers that are cooked prior to being passed to the console.

 
## References

* [x] I've discussed this with core contributors already (internal)

## Validation Steps Performed
- Ensured trace is correctly logged in ETL (via TraceLog)

### CODE REVIEW

File: src/host/tracing.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@ enum TraceKeywords
     Input = 0x200,
     API = 0x400,
     UIA = 0x800,
+    CookedRead = 0x1000,
     All = 0xFFF
```
Comment (miniksa) [inline src/host/tracing.cpp:24]: Update all to either 0x1FFF or just 0xFFFF for future proof.

Comment (depletionmode) [reply src/host/tracing.cpp:24]: Whoops. yes

Comment (depletionmode) [reply src/host/tracing.cpp:24]: I think 1fff better aligns with the comment in til.h

File: src/host/readDataCooked.cpp [hunk 2]
```diff
@@ -1028,6 +1028,8 @@ void COOKED_READ_DATA::SavePendingInput(const size_t index, const bool multiline
                 CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
                 LOG_IF_FAILED(_commandHistory->Add({ _backupLimit, StringLength / sizeof(wchar_t) },
                                                    WI_IsFlagSet(gci.Flags, CONSOLE_HISTORY_NODUP)));
+
+                Tracing::s_TraceCookedRead(_backupLimit);
```
Comment (miniksa) [inline src/host/readDataCooked.cpp:1032]: If there are too many client processes attached, some of them will not have a history list. We may want to place this outside the `if (_commandHistory)` block.

Comment (depletionmode) [reply src/host/readDataCooked.cpp:1032]: Ah. Didn't realise that could be a thing. Makes sense will address.

**Review by miniksa — APPROVED** [review]
Looks good to me!

**Review by zadjii-msft — APPROVED** [review]
lgtm

**Review by DHowett — APPROVED** [review]

**Pull request merged**